pontoon/terminology/models.py

```diff
@@ -0,0 +1,251 @@
+                except ValueError:
+                    break
+
+        return terms_phrases.items()
+
+    @transaction.atomic
+    def assign_terms_to_entities(self, entities):
+        entity_terms = []
+        entities_pks = [e[0] for e in entities]
+        # Remove existing terms relations.
+        EntityTerm.objects.filter(entity__in=entities_pks).delete()
+        for entity_pk, string, string_singulars, string_plural, string_plural_singulars in entities:
+            if string_singulars:
+                entity_terms.extend([
+                    (entity_pk, term_pks, phrase)
+                    for (phrase, term_pks) in self.find_terms(string, string_singulars)
+                ])
+
+            if string_plural_singulars:
+                entity_terms.extend([
+                    (entity_pk, term_pks, phrase)
+                    for (phrase, term_pks) in self.find_terms(
+                        string_plural, string_plural_singulars)
+                ])
+        terms_db = []
+
+        for entity_pk, terms, phrase in entity_terms:
+            for term_pk in terms:
+                terms_db.append(
+                    EntityTerm(entity_id=entity_pk, term_id=term_pk, phrase=phrase)
+                )
+        EntityTerm.objects.bulk_create(terms_db, 10000)
+
+        terms_count = (
+            Term.objects.filter(entityterm__entity__pk__in=entities_pks)
+            .distinct()
+            .count()
+        )
+
+        phrases_count = (
+            EntityTerm.objects.filter(entity__pk__in=entities_pks)
+            .distinct()
+            .count()
+        )
+
+        log.info(
+            'Assigned %s terms to %s entities and matched %s phrases.',
+            terms_count,
+            len(entities),
+            phrases_count
+        )
+
+    @transaction.atomic
+    def import_tbx_terms(self, tbx_terms):
+        """
+        Creates/updates term objects passed from terminology files.
+        All terms will be automatically matched with their entities.
+
+        :param VCSTerm tbx_terms: a list of terms imported from the filesystem/vcs.
+        """
+        db_terms = {}
+
+        for term in Term.objects.all():
+            db_terms[term.term_id] = term
+
+        new_terms = []
+        update_terms = []
+        updated_records = 0
+
+        for term_id, term in tbx_terms.items():
+            try:
+                db_term = db_terms[term.term_id]
+                db_term.part_of_speech = term.part_of_speech
+                db_term.description = term.description
+                update_terms.append(db_term)
+                updated_records += 1
+            except KeyError:
+                new_terms.append(
+                    Term(
+                        term_id=term.term_id,
+                        source_term=term.source_text,
+                        source_term_singulars=' '.join(get_singulars(term.source_text)),
+                        note=term.part_of_speech or '',
```

Comment:
I don't understand, why is this called `note` in one case and `part_of_speech` in another?